be/src/pipeline/exec/olap_scan_operator.cpp

```diff
@@ -410,20 +388,67 @@ Status OlapScanLocalState::_init_scanners(std::list<vectorized::VScannerSPtr>* s
                                   state(),
                                   _scanner_profile.get(),
                                   scanner_ranges,
-                                  tablet,
+                                  _tablets[scan_range_idx].tablet,
                                   version,
-                                  {},
+                                  _read_sources[scan_range_idx],
                                   p._limit,
                                   p._olap_scan_node.is_preaggregation,
                           });
             RETURN_IF_ERROR(scanner->prepare(state(), _conjuncts));
             scanners->push_back(std::move(scanner));
         }
     }
+    _tablets.clear();
+    _read_sources.clear();
 
     return Status::OK();
 }
 
+Status OlapScanLocalState::hold_tablets() {
```

Comment:
if the function cost too mush time, should log warning in BE. the clould mode may have io in pipeline thread may hang the thread, it's danger way.